uk_bin_collection/uk_bin_collection/councils/CrawleyBoroughCouncil.py

```
@@ -86,27 +84,69 @@ def parse_data(self, page: str, **kwargs) -> dict:
+                                bin_type = key.replace("DateNext", "")
+                            next_collection = datetime.strptime(value.strip(), "%A %d %B").replace(
+                                year=datetime.now().year
+                            )
+                            if datetime.now().month == 12 and next_collection.month == 1:
+                                next_collection = next_collection + relativedelta(years=1)
+                            bindata["bins"].append({
+                                "type": bin_type,
+                                "collectionDate": next_collection.strftime(date_format),
+                            })
+            except ET.ParseError as e:
+                raise ValueError(f"Failed to parse XML response: {e}")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)